### ISSUE TITLE
GZIP compression impossible?

### ISSUE THREAD

**Issue by pkucmus** [aws/chalice#1776]
**Problem:**
Probably something HTTPS related is being done to the body of the response only once deployed, a local service does not have this issue (which might be another issue - that there are discrepancies so big between local and the real thing). Trying to follow the https://aws.github.io/chalice/tutorials/basicrestapi.html?highlight=gzip#gzip-compression-for-json I endup with a `UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte` ONLY on a deployed lambda. `0x8b` is a part of GZIP's [magic number](https://en.wikipedia.org/wiki/List_of_file_signatures).

**Steps to reproduce:**
Given the example from: https://aws.github.io/chalice/tutorials/basicrestapi.html?highlight=gzip#gzip-compression-for-json

```python
import json
import gzip
from chalice import Chalice, Response

app = Chalice(app_name='compress-response')
app.api.binary_types.append('application/json')

@app.route('/')
def index():
    blob = json.dumps({'hello': 'world'}).encode('utf-8')
    payload = gzip.compress(blob)
    custom_headers = {
        'Content-Type': 'application/json',
        'Content-Encoding': 'gzip'
    }
    return Response(body=payload,          # this is bytes
                    status_code=200,
                    headers=custom_headers)
```
1. deploy it with `chalice local` - all is good
2. deploy with `chalice deploy` - you will get the `UnicodeDecodeError`

**Expected behavior:**
Both instances should work the same - respond with a gzip compressed response.

This is potentially related to #1666 

Thank you for looking into this.

**Comment by pkucmus** [comment]
The issue might be of a different nature, it seems that once the instance in deployed I'm running into double compression / decompression as AWS does that on the gateway level: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-gzip-compression-decompression.html and https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-enable-compression.html#api-gateway-supported-content-encodings

Maybe the example from docs needs to go away (or at least describe why it's not a prod example) and additionally we could use a web proxy (like nginx) in front of a local deployment that would mimic what AWS does before a requests reaches a lambda?
